### ISSUE TITLE
Serialization support

### ISSUE THREAD

**Issue by kyrias** [michaelsproul/rust_sequence_trie#38]
This crate is pretty much exactly what I could really use for a project of mine, but I'd need to easily be able to serialize it to disk in a portable way.  Is there any chance of a serde serialization implementation happening?

**Comment by michaelsproul** [comment]
Yeah, it's very much on the cards! My other trie (`radix_trie`) has a Serde serialisation implementation that someone contributed, and we could probably copy much of the code from there.

I'm on vacation at the moment, and probably wouldn't get around to it for 2-3 weeks. Do you feel like giving it a shot in the meantime?

**Comment by kyrias** [comment]
I'm not sure if I'm knowledgeable enough to do it, but I'll see if I can give it a shot this weekend.  Have a nice vacation!

**Comment by kyrias** [comment]
Actually seem to have got it working by copying the `radix_trie` serde implementation, and doing some small changes, but one problem is that the `HashMap` iteration order is arbitrary, so the tests are currently failing most of the time.

Would need to somehow sort the key iterator, hm.

**Comment by kyrias** [comment]
Ah, I can just pass the same instance of `RandomState` to `::with_hasher`.

**Comment by kyrias** [comment]
So, I worked around the `HashMap` iteration order problem by switching to `BTreeMap`.  WIP `serde` branch is available [here](https://github.com/kyrias/rust_sequence_trie).

**Comment by kyrias** [comment]
... And after fighting with Serde to get the deserialization implementaiton working for a few hours, I now realize that the auto-derived ones work just fine.  *sigh*.

**Comment by michaelsproul** [comment]
The `HashMap` option is mostly for speed. Thanks for looking into this! I'm back from vacation now so I should have some time to look into it soon
